base/linalg/triangular.jl

```diff
@@ -1547,12 +1547,11 @@ for (f, g) in ((:A_mul_Bc, :A_mul_Bc!), (:A_mul_Bt, :A_mul_Bt!))
     end
 end
 
-for mat in (:AbstractVector, :AbstractMatrix)
-
 ### Multiplication with triangle to the left and hence rhs cannot be transposed.
 for (f, g) in ((:*, :A_mul_B!), (:Ac_mul_B, :Ac_mul_B!), (:At_mul_B, :At_mul_B!))
     @eval begin
-        function ($f)(A::AbstractTriangular, B::$mat)
+        ($f)(A::AbstractTriangular, B::AbstractVector) = ($f)(A, reshape(B, Val{2}))
```

Comment:
Echoing `@andreasnoack`, these changes result in a matrix being returned rather than a vector, e.g.
```julia
julia> UpperTriangular(rand(3, 3)) * rand(3)
3×1 Array{Float64,2}:
 0.978125
 0.687285
 0.568977
```
Best!